src/options.rs

```diff
@@ -51,14 +38,21 @@ pub struct Options {
     pub block_restart_interval: usize,
     /// Note: you have to open a database with the same compression type as it was written to, in
     /// order to not lose data! (this is a bug and will be fixed)
-    pub compression_type: CompressionType,
+    pub compressor: u8,
+
+    pub compressor_list: Rc<[Option<Box<dyn Compressor>>; 256]>,
```

Comment:
this is a good idea, but could you encapsulate this? for example with an API like

```rust
let mut compressor_list = CompressorList::new();
compressor_list.set(/*type =*/0, NoneCompressor::new());
compressor_list.set(1, SnappyCompressor::new());
let c = compressor_list.get(1);
```

That would look a bit nicer to users of the `Options` type supplying their own compressors.

Reply:
You are right! It look nicer and cleaner in user api and internal code.